c10/core/SymBool.cpp

```diff
@@ -69,4 +69,11 @@ bool SymBool::guard_bool(const char* file, int64_t line) const {
   return a->guard_bool(file, line);
 }
 
+bool SymBool::has_hint() const {
+  if (!is_symbolic()) {
+    return true;
+  }
+  return toSymNodeImpl()->has_hint();
```

Comment:
Nit: Positive conditions for early returns feel more readable

```
if (is_symbolic()) {
      return toSymNodeImpl()->has_hint();
  }
return false;
```

Also, j/w - how can a symbool be not symbolic?

Reply:
Swapping the condition would be inconsistent with all of the other SymInt/SymFloat implementations (also, Nick was the one who picked the convention, not me.) The alternate rule which makes the early return make sense to me is the not symbolic case tends to be the "easy" case (since you just run the regular operation), and the symbolic case is the complicated one (so you don't want to indent it.)